i6_models/config.py

```diff
@@ -51,3 +53,22 @@ def _validate_types(self) -> None:
 
     def __post_init__(self) -> None:
         self._validate_types()
+
+
+ConfigType = TypeVar("ConfigType", bound=ModelConfiguration)
+ModuleType = TypeVar("ModuleType", bound=nn.Module)
+
+
+@dataclass
+class SubassemblyFactory(Generic[ConfigType, ModuleType]):
```

Comment:
Should this be versionized?
```suggestion
class SubassemblyFactoryV1(Generic[ConfigType, ModuleType]):
```

Reply:
I dont think so. This is no part/assembly/primitive

Reply:
So only parts/assemblies/primitives should be versionized, and everything else not? Why this inconsistency?

Reply:
I don't have a strong opinion on this, but I think we don't expect this module to change, since its quite basic.